File: TypeScript/ProjectEuler1.ts
class RangeList {
    private range: number[] = [];

    constructor(from: number, to: number) {
        for (var i = from; i < to; i++) {
            this.range.push(i);
        }
    }
    
    toList() {
        return this.range;
    }
}

var main = function() {
    var list: number[] = new RangeList(1, 1000).toList();
    return list.filter(x => x % 5 === 0 || x % 3 === 0).reduce((x,y)=>x + y);
}

console.log(main());

File: Scala/ProjectEuler1.scala
class Testy {
  def sumListElements(data:List[Int]):Int = {
    return data.filter(x => x % 3 == 0 || x % 5 == 0).foldLeft(0)(_+_)
  }
}

val a = new Testy().sumListElements(List.range(1,1000))

a
